lib/pragma-grammar.ym

```diff
@@ -161,6 +163,12 @@ module_param
 	: LL_IDENTIFIER '(' string_or_number ')'		{ cfg_args_set(last_module_args, $1, $3); free($1); free($3); }
 	;
 
+config_id_stmt
+  : KW_CONFIG_ID ':' positive_integer64
+    {
+      configuration->user_config_id = $3;
```

Comment:
Don't we want to allow something like a git commit id? Or other version number like value?

Reply:
I can see that Prometheus does not support non-numeric values, so I understand this is numeric too, if we want to expose this value as a metric.

But do we need to export this as a metric?  I understand that we want to see if there was an issue with config reload, but we already have metrics for successful reloads and reloads in general, and if the two differs it means that the config wasn't successfully reloaded. 

We could potentially expose the timestamp of the config file, that should indicate that a reload didn't even start.

If the management service in question needs the config id, it can get it in other ways (e.g. running syslog-ng-ctl config --id)

One more idea is to include the config id as a label to the successful reload metric, although that would mean we would profilerate metrics, as we would have a separate one for each reload operation.

Reply:
We could allow strings, but in that case we can't publish this as a metric.

Reply:
Good idea. I'll remove the config-id from metrics (allowing strings), and publish the main configuration's modification timestamp as a metric.

(Adding the config-id as a label does not sound like the "Prometheus way to go", label cardinality highly affects the performance of Prometheus, and the metric itself would be even more difficult to query/understand.)

Reply:
Agreed. Although I think that having a few reloads per day shouldn't be that much of a problem and Prometheus would eventually garbage collect old counters that don't receive updates. But I am also on an opinion that if the management system is in a position to trigger a reload, it can also query the current config version.

During reconsiliation, the comparison between desired state vs. current state would likely use this logic.